native/KotlinIOS/KotlinIOS/ViewController.swift

```diff
@@ -28,13 +42,28 @@ class ViewController: UIViewController, ApplicationContractView {
         
         setUpPickers()
         setUpTable()
+        setUpTimePicker()
     }
     
     
-    @IBAction func onButtonTapped() {
+    @IBAction func submitButtonTapped() {
         let originStation = stations[departurePicker.selectedRow(inComponent:0)]
         let finalStation = stations[arrivalPicker.selectedRow(inComponent:0)]
-        presenter.makeTrainSearch(originCrs: originStation, destinationCrs: finalStation)
+        let dateTime = convertDateToCorrectFormat(date: datePicker.date)
+        presenter.makeTrainSearch(originCrs: originStation, destinationCrs: finalStation, dateTime: dateTime)
+    }
+}
+
+extension ViewController {
+    func convertDateToCorrectFormat(date: Date) -> String {
+        let dateString = date.description
+        let date = dateString.prefix(10)
+        
+        let start = dateString.index(dateString.startIndex, offsetBy: 11)
+        let end = dateString.index(dateString.startIndex, offsetBy: 18)
+        let time = String(dateString[start...end])
+        
+        return date + "T" + time + ".000+00:00"
     }
```

Comment:
Again can we handle this (and the time functions below) differently? And is there a way to pull out handling to the core at all?